job/runner_remote_param.go

```diff
@@ -0,0 +1,214 @@
+package job
+
+import (
+	"encoding/json"
+	"io"
+	"net/url"
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+const (
+	mimeFormURLEncoded = "application/x-www-form-urlencoded"
+	mimeFormData       = "multipart/form-data"
+)
+
+var (
+	typeOfPointerToBinaryData = reflect.TypeOf(&BinaryData{})
+	typeOfPointerToBinaryFile = reflect.TypeOf(&BinaryFile{})
+)
+
+// BinaryData represents binary data from a given source.
+type BinaryData struct {
+	Filename    string    // filename used in multipart form writer.
+	Source      io.Reader // file data source.
+	ContentType string    // content type of the data.
+}
+
+// BinaryFile represents a file on disk.
+type BinaryFile struct {
+	Filename    string // filename used in multipart form writer.
+	Path        string // path to file. must be readable.
+	ContentType string // content type of the file.
+}
+
+// Params is the params used to send to API server.
+//
+// For general uses, just use Params as an ordinary map.
+//
+// For advanced uses, use MakeParams to create Params from any struct.
+type Params map[string]interface{}
```

Comment:
I don't totally understand the purpose of this type.

1. It doesn't seem to have much use in Kala itself.
2. It seems like a utility type for a client to encode data into the body of a RemoteJob.

Is that a correct evaluation?  If so, I think it may be out of scope for Kala.

Reply:
```curl -X POST \
  -F 'name="My new Custom Audience"' \
  -F 'subtype="CUSTOM"' \
  -F 'description="People who purchased on my website"' \
  -F 'customer_file_source="USER_PROVIDED_ONLY"' \
  -F 'access_token=<ACCESS_TOKEN>' \
  https://graph.facebook.com/v7.0/act_<AD_ACCOUNT_ID>/customaudiences```
in case we need to send multipart request like Facebook GraphAPI! current implementation doesn't allow us to prepare request like above!
by having Params, we can reconstruct request for `-F` parameters.